### PR TITLE
Modified themeing with constants for colors

### PR DESCRIPTION
**Pull request by devYaoYH**
- Modified `theme.ts` to declare and use constants for colors
- Added shadow to ItemCardMedia
- Renamed `ErrorTheme` -> `LightErrorTheme` since the background is white and text has success/error color instead
- Added `ErrorTheme` to move 'success' and 'error' theme options into 'primary' and 'secondary' respectively to use in typescript-compliant @material-ui `color=...` prop field
- Added debug toggle switch to quickly check layout view in 'prod' without debugging controls
Screenshot 2020-07-01 at 13 15 14

### CODE REVIEW

File: client/src/theme.ts [hunk 1]
```diff
@@ -1,38 +1,62 @@
...
+      contrastText: WHITE_TEXT_COLOR,
+    },
+    secondary: {
+      main: ERROR_COLOR,
+      contrastText: WHITE_TEXT_COLOR,
+    },
+  },
+});
+
+// Inverse of ErrorTheme with text-color and backgrouund color inverted
```
Comment (BhavyaLight) [inline client/src/theme.ts:50]: is this for dark mode?

Comment (devYaoYH) [reply client/src/theme.ts:50]: no, this is 'default' mode
Screenshot 2020-07-01 at 15 35 11
currently, only 'primary' and 'secondary' are valid options for the `color` prop in most material-ui components, but perhaps at least we need the 'success' and 'error' ones too, I'm just moving those colors into the 'primary' and 'secondary' fields and wrapping a different `<MuiThemeProvider theme={} />` to allow us to configure these consistent color schemes and use them in components

Comment (BhavyaLight) [reply client/src/theme.ts:50]: gotcha! So whats the difference between LightErrorTheme & ErrorTheme in terms of usage?

Comment (devYaoYH) [reply client/src/theme.ts:50]: LightErrorTheme has the colors flipped around, where the text color is 'success' or 'error' and main color's 'white' instead, eh, perhaps `InvertedErrorTheme` is better?

**Pull request merged**